cmd/run.go

```diff
@@ -36,7 +36,14 @@ func newRunCmd(opts *lefthook.Options) *cobra.Command {
 		"run hooks on all files",
 	)
 
-	runCmd.Flags().StringSliceVar(&runArgs.Files, "files", nil, "run on specified files. takes precedence over --all-files")
+	runCmd.Flags().BoolVar(
+		&runArgs.FilesFromStdin, "files-from-stdin", false,
+		"get files from standard input, null-separated, instead of from CLI parameters --file(s)",
+	)
+
+	runCmd.Flags().StringSliceVar(&runArgs.Files, "files", nil, "[DEPRECATED] run on specified files, comma-separated. takes precedence over --all-files")
```

Comment:
Please, refer this [this code example](https://github.com/evilmartians/lefthook/blob/master/cmd/install.go#L31) to mark deprecations.

Reply:
Done in [ea12589](https://github.com/evilmartians/lefthook/pull/596/commits/ea125897aa68cbeec62b9b21bf99e633d24931b2).